client/src/container/App.js

```diff
@@ -0,0 +1,162 @@
+import mock from "../mock.json"
+
+
+class App extends Component {
+
+    constructor() {
+        super();
+        this.state = {
+            term: '',
+            searchBy: 'title',
+            sortBy: 'release date',
+            searched: false,
+            allFilms: [],
+            activeAlbumId: null,
+            activeFilm: null,
+            someGenreFilms: []
+
+        };
+        this.toggleSearchBy = this.toggleSearchBy.bind(this);
+        this.formSubmitHandler = this.formSubmitHandler.bind(this);
+        this.inputChangeHandler = this.inputChangeHandler.bind(this);
+        this.toggleSortBy = this.toggleSortBy.bind(this);
+        this.searchFilms = this.searchFilms.bind(this);
+        this.albumClickHandler = this.albumClickHandler.bind(this);
+        this.searchButtonHandler = this.searchButtonHandler.bind(this);
+    }
+
+    toggleSearchBy(e) {
+        this.setState({
+            searchBy: e.target.value
+        });
+    };
+
+    inputChangeHandler(e) {
+        this.setState({
+            term: e.target.value
+        })
+    };
+
+    formSubmitHandler(e) {
+        e.preventDefault();
+        this.searchFilms(this.state.searchBy, this.state.term.toLowerCase());
+        this.setState({
+            term: '',
+            searched: true
+        });
+
+    };
+
+    toggleSortBy(e) {
+        let sortFunction = this.state.sortBy === 'release date' ?
+            (a, b) => {
+                return new Date(a['release_date']) - new Date(b['release_date']);
+            } :
+            (a, b) => {
+                return a['vote_average'] - b['vote_average']
+            };
+
+        this.setState({
+            sortBy: e.target.value,
+            allFilms: this.state.allFilms.sort(sortFunction),
+        })
+    };
+
+    searchFilms(searchBy, term) {
+        let filterVariable = searchBy === 'title' ? 'title' : 'genres';
+        let filteredFilms = mock.data.filter((item) => {
+            if (typeof item[filterVariable] === 'string') {
+                return item[filterVariable].toLowerCase().indexOf(term) > -1;
+            }
+            if (typeof item[filterVariable] === 'object') {
+                return item[filterVariable].some((item) => {
+                    return item.toLowerCase().indexOf(term) > -1
+                })
+            }
+        });
+
+        this.setState({
+            allFilms: filteredFilms
+        });
+    }
+
+    albumClickHandler(id) {
+        let targetFilm = this.state.allFilms.filter((film) => {
+            return film.id === id
+        })[0];
+
+        this.setState({
+            activeAlbumId: id,
+            activeFilm: targetFilm
+        })
+    }
+
+    searchButtonHandler() {
+        this.setState({
+            activeAlbumId: null,
+            activeFilm: null
+        });
+    }
+
+    render() {
+        return (
+            <AppWrapper>
+                <ErrorBoundary>
+                    <Header>
```

Comment:
Consider making this component not only a style-component, but a container with its own methods and state.
It'll reduce the code in App.js